libs/langchain/langchain/chat_models/anyscale.py

```diff
@@ -0,0 +1,149 @@
+"""Anyscale Endpoints chat wrapper. Relies heavily on ChatOpenAI."""
+from __future__ import annotations
+
+import logging
+import sys
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from langchain.chat_models.openai import (
+    _convert_message_to_dict,
+    _import_tiktoken,
+    ChatOpenAI,
+)
+from langchain.schema.messages import BaseMessage
+from langchain.utils import get_from_dict_or_env
+from pydantic import Field, root_validator
+
+if TYPE_CHECKING:
+    import tiktoken
+
+logger = logging.getLogger(__name__)
+
+
+class AnyscaleChatModel(str, Enum):
```

Comment:
Currently, these are the models that endpoints support, but the plan is to add more. Should we use the /models endpoint to get a list rather than have it static? 

OK to leave it as is for now, but let's leave a todo to make it dynamic in future?

Reply:
Here's an example of a curl command to get the list of models (with metadata): 
```
curl "$OPENAI_API_BASE/models" \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $OPENAI_API_KEY"
```

Reply:
I definitely think we should design for the future. I'm not exactly sure how to make that work with the enum. Maybe we need a different mechanism. A Todo would be fine, but maybe fixing it now would be preferable. 

Any ideas on proper implementation would be welcomed; otherwise, I will submit a new commit in the next few days.

Thank you :)

Reply:
I removed the enum. Instead, the root validator raises a ValueError if model_name is invalid according to the `/models` endpoint.